### PR TITLE
Add StrictModeRule for use by SDKs.

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: integ-testing/src/main/java/com/google/firebase/testing/integ/StrictModeRule.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                  StrictMode.setVmPolicy(createVmPolicy(violations));
+                });
+        try {
+          base.evaluate();
+        } catch (Throwable e) {
+          violations.add(e);
+        } finally {
+          InstrumentationRegistry.getInstrumentation()
+              .runOnMainSync(() -> StrictMode.setThreadPolicy(originalThreadPolicy.get()));
+          runGc();
```
Comment (rlazo) [inline integ-testing/src/main/java/com/google/firebase/testing/integ/StrictModeRule.java:132]: out of curiosity, why is this part needed?

Comment (vkryachko) [reply integ-testing/src/main/java/com/google/firebase/testing/integ/StrictModeRule.java:133]: added a comment with an explanation.

**Review by rlazo — APPROVED** [review]
